src/Nethermind/Nethermind.JsonRpc/Modules/RpcModuleAttribute.cs

```diff
@@ -0,0 +1,31 @@
+//  Copyright (c) 2021 Demerzel Solutions Limited
+//  This file is part of the Nethermind library.
+// 
+//  The Nethermind library is free software: you can redistribute it and/or modify
+//  it under the terms of the GNU Lesser General Public License as published by
+//  the Free Software Foundation, either version 3 of the License, or
+//  (at your option) any later version.
+// 
+//  The Nethermind library is distributed in the hope that it will be useful,
+//  but WITHOUT ANY WARRANTY; without even the implied warranty of
+//  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+//  GNU Lesser General Public License for more details.
+// 
+//  You should have received a copy of the GNU Lesser General Public License
+//  along with the Nethermind. If not, see <http://www.gnu.org/licenses/>.
+// 
+
+using System;
+
+namespace Nethermind.JsonRpc.Modules
+{
+    public class RpcModuleAttribute : Attribute
+    {
```

Comment:
created but not used in the end?
I have been thinking about this for a long time
never liked the ModuleType enum so glad that there is an attempt to remove it
I remember it was causing some issues though in the ModuleLoader (when resolving rpc method we check whether the module is enabled or unknown and define the error message based on it)

Reply:
It was just moved to separate file. Modules can be registered by string now.